### PR TITLE
Add iPhone 14 Devices via new SnapshotTestingExtensions target

### PR DESCRIPTION
**Pull request by tinder-garricnahapetian**

### CODE REVIEW

File: Tests/LayoutTests/Support/UITraitCollection.swift [hunk 1]
```diff
@@ -0,0 +1,114 @@
+//
+//  All Contributions by Match Group
+//
+//  Copyright © 2023 Tinder (Match Group, LLC)
```
Comment (tinder-cfuller) [inline Tests/LayoutTests/Support/UITraitCollection.swift:4]:
```suggestion
//  Copyright © 2024 Tinder (Match Group, LLC)
```

File: Tests/LayoutTests/Support/UITraitCollection.swift [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                    UITraitCollection(verticalSizeClass: .compact)
+                ]
+            )
+        case .portrait:
+            return UITraitCollection(
+                traitsFrom: base + [
+                    UITraitCollection(horizontalSizeClass: .compact),
+                    UITraitCollection(verticalSizeClass: .regular)
+                ]
+            )
```
Comment (tinder-cfuller) [inline Tests/LayoutTests/Support/UITraitCollection.swift:36]: Please refactor all to this style. Thank you!
```suggestion
        return switch orientation {
        case .landscape:
            UITraitCollection(traitsFrom: base + [
                UITraitCollection(horizontalSizeClass: .regular),
                UITraitCollection(verticalSizeClass: .compact)
            ])
        case .portrait:
            UITraitCollection(traitsFrom: base + [
                UITraitCollection(horizontalSizeClass: .compact),
                UITraitCollection(verticalSizeClass: .regular)
            ])
```

File: Tests/LayoutTests/Support/ViewImageConfig.swift [hunk 3]
```diff
@@ -0,0 +1,75 @@
+//
+//  All Contributions by Match Group
+//
+//  Copyright © 2023 Tinder (Match Group, LLC)
```
Comment (tinder-cfuller) [inline Tests/LayoutTests/Support/ViewImageConfig.swift:4]:
```suggestion
//  Copyright © 2024 Tinder (Match Group, LLC)
```

File: Tests/LayoutTests/Support/ViewImageConfig.swift [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+import SnapshotTesting
+import UIKit
+
+extension ViewImageConfig {
+
+    internal static let iPhone14: ViewImageConfig = .iPhone14(.portrait)
+    internal static let iPhone14Plus: ViewImageConfig = .iPhone14Plus(.portrait)
+    internal static let iPhone14Pro: ViewImageConfig = .iPhone14Pro(.portrait)
+    internal static let iPhone14ProMax: ViewImageConfig = .iPhone14ProMax(.portrait)
```
Comment (tinder-cfuller) [inline Tests/LayoutTests/Support/ViewImageConfig.swift:18]: If these are not used, please remove.
```suggestion
```

**Review by tinder-cfuller — COMMENTED** [review]
May you please create a new library product named `SnapshotTestingExtensions` to contain the new extensions? They will need to be made public in this case. The reason for this suggestion is so that other consumers may use the new device sizes as well. Thank you!

File: Package.swift [hunk 5]
```diff
@@ -23,11 +26,18 @@ let package = Package(
...
+            swiftSettings: .swiftSettings,
+            plugins: [
+                .plugin(name: SwiftLint.plugin),
+            ]),
         .testTarget(
             name: "LayoutTests",
             dependencies: [
                 "Layout",
                 "Nimble",
+                "SnapshotTestingExtensions",
```
Comment (tinder-cfuller) [inline Package.swift:40]: Place immediately following `Layout` so it is together with the other target in this package.
```suggestion
                "Layout",
                "SnapshotTestingExtensions",
                "Nimble",
```

File: Package@swift-5.10.swift [hunk 6]
```diff
@@ -34,11 +37,18 @@ let package = Package(
...
+            swiftSettings: .swiftSettings,
+            plugins: [
+                .plugin(name: SwiftLint.plugin),
+            ]),
         .testTarget(
             name: "LayoutTests",
             dependencies: [
                 "Layout",
                 "Nimble",
+                "SnapshotTestingExtensions",
```
Comment (tinder-cfuller) [inline Package@swift-5.10.swift:51]: Place immediately following `Layout` so it is together with the other target in this package.
```suggestion
                "Layout",
                "SnapshotTestingExtensions",
                "Nimble",
```

File: Package@swift-5.9.swift [hunk 7]
```diff
@@ -34,11 +37,18 @@ let package = Package(
...
+            swiftSettings: .swiftSettings,
+            plugins: [
+                .plugin(name: SwiftLint.plugin),
+            ]),
         .testTarget(
             name: "LayoutTests",
             dependencies: [
                 "Layout",
                 "Nimble",
+                "SnapshotTestingExtensions",
```
Comment (tinder-cfuller) [inline Package@swift-5.9.swift:51]: Place immediately following Layout so it is together with the other target in this package.
```suggestion
                "Layout",
                "SnapshotTestingExtensions",
                "Nimble",
```

File: Sources/SnapshotTestingExtensions/UITraitCollection.swift [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+            UITraitCollection(userInterfaceIdiom: .phone)
+        ]
+        return switch orientation {
+        case .landscape:
+            UITraitCollection(
+                traitsFrom: base + [
+                    UITraitCollection(horizontalSizeClass: .regular),
+                    UITraitCollection(verticalSizeClass: .compact)
+                ]
+            )
```
Comment (tinder-cfuller) [inline Sources/SnapshotTestingExtensions/UITraitCollection.swift:29]: May we try this alternate style?
```suggestion
            UITraitCollection(traitsFrom: base + [
                UITraitCollection(horizontalSizeClass: .regular),
                UITraitCollection(verticalSizeClass: .compact)
            ])
```

File: Sources/SnapshotTestingExtensions/UITraitCollection.swift [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+        case .portrait:
+            UITraitCollection(
+                traitsFrom: base + [
+                    UITraitCollection(horizontalSizeClass: .compact),
+                    UITraitCollection(verticalSizeClass: .regular)
+                ]
+            )
```
Comment (tinder-cfuller) [inline Sources/SnapshotTestingExtensions/UITraitCollection.swift:36]: May we try this alternate style?
```suggestion
            UITraitCollection(traitsFrom: base + [
                UITraitCollection(horizontalSizeClass: .compact),
                UITraitCollection(verticalSizeClass: .regular)
            ])
```

File: Sources/SnapshotTestingExtensions/UITraitCollection.swift [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+            UITraitCollection(userInterfaceIdiom: .phone)
+        ]
+        return switch orientation {
+        case .landscape:
+            UITraitCollection(
+                traitsFrom: base + [
+                    UITraitCollection(horizontalSizeClass: .regular),
+                    UITraitCollection(verticalSizeClass: .compact)
+                ]
+            )
```
Comment (tinder-cfuller) [inline Sources/SnapshotTestingExtensions/UITraitCollection.swift:54]: May we try this alternate style?
```suggestion
            UITraitCollection(traitsFrom: base + [
                UITraitCollection(horizontalSizeClass: .regular),
                UITraitCollection(verticalSizeClass: .compact)
            ])
```

File: Sources/SnapshotTestingExtensions/UITraitCollection.swift [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+        case .portrait:
+            UITraitCollection(
+                traitsFrom: base + [
+                    UITraitCollection(horizontalSizeClass: .compact),
+                    UITraitCollection(verticalSizeClass: .regular)
+                ]
+            )
```
Comment (tinder-cfuller) [inline Sources/SnapshotTestingExtensions/UITraitCollection.swift:61]: May we try this alternate style?
```suggestion
            UITraitCollection(traitsFrom: base + [
                UITraitCollection(horizontalSizeClass: .compact),
                UITraitCollection(verticalSizeClass: .regular)
            ])
```

File: Sources/SnapshotTestingExtensions/UITraitCollection.swift [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+            UITraitCollection(userInterfaceIdiom: .phone)
+        ]
+        return switch orientation {
+        case .landscape:
+            UITraitCollection(
+                traitsFrom: base + [
+                    UITraitCollection(horizontalSizeClass: .regular),
+                    UITraitCollection(verticalSizeClass: .compact)
+                ]
+            )
```
Comment (tinder-cfuller) [inline Sources/SnapshotTestingExtensions/UITraitCollection.swift:79]: May we try this alternate style?
```suggestion
            UITraitCollection(traitsFrom: base + [
                UITraitCollection(horizontalSizeClass: .regular),
                UITraitCollection(verticalSizeClass: .compact)
            ])
```

File: Sources/SnapshotTestingExtensions/UITraitCollection.swift [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+        case .portrait:
+            UITraitCollection(
+                traitsFrom: base + [
+                    UITraitCollection(horizontalSizeClass: .compact),
+                    UITraitCollection(verticalSizeClass: .regular)
+                ]
+            )
```
Comment (tinder-cfuller) [inline Sources/SnapshotTestingExtensions/UITraitCollection.swift:86]: May we try this alternate style?
```suggestion
            UITraitCollection(traitsFrom: base + [
                UITraitCollection(horizontalSizeClass: .compact),
                UITraitCollection(verticalSizeClass: .regular)
            ])
```

File: Sources/SnapshotTestingExtensions/UITraitCollection.swift [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+            UITraitCollection(userInterfaceIdiom: .phone)
+        ]
+        return switch orientation {
+        case .landscape:
+            UITraitCollection(
+                traitsFrom: base + [
+                    UITraitCollection(horizontalSizeClass: .regular),
+                    UITraitCollection(verticalSizeClass: .compact)
+                ]
+            )
```
Comment (tinder-cfuller) [inline Sources/SnapshotTestingExtensions/UITraitCollection.swift:104]: May we try this alternate style?
```suggestion
            UITraitCollection(traitsFrom: base + [
                UITraitCollection(horizontalSizeClass: .regular),
                UITraitCollection(verticalSizeClass: .compact)
            ])
```

File: Sources/SnapshotTestingExtensions/UITraitCollection.swift [hunk 15]
```diff
@@ -0,0 +1,114 @@
...
+        case .portrait:
+            UITraitCollection(
+                traitsFrom: base + [
+                    UITraitCollection(horizontalSizeClass: .compact),
+                    UITraitCollection(verticalSizeClass: .regular)
+                ]
+            )
```
Comment (tinder-cfuller) [inline Sources/SnapshotTestingExtensions/UITraitCollection.swift:111]: May we try this alternate style?
```suggestion
            UITraitCollection(traitsFrom: base + [
                UITraitCollection(horizontalSizeClass: .compact),
                UITraitCollection(verticalSizeClass: .regular)
            ])
```

File: Tests/LayoutTests/Support/SnapshotTesting.swift [hunk 16]
```diff
@@ -10,6 +10,7 @@
 
 import Nimble
 import SnapshotTesting
+import SnapshotTestingExtensions
```
Comment (tinder-cfuller) [inline Tests/LayoutTests/Support/SnapshotTesting.swift:13]: If this is not needed, please remove.
```suggestion
```

File: Package.swift [hunk 17]
```diff
@@ -23,11 +26,18 @@ let package = Package(
     targets: [
         .target(
             name: "Layout"),
+        .target(
+            name: "SnapshotTestingExtensions",
```
Comment (tinder-cfuller) [inline Package.swift:30]:
```suggestion
            name: "SnapshotTestingExtensions",
            dependencies: [
                .product(name: "SnapshotTesting", package: "swift-snapshot-testing"),
            ],
```

File: Package@swift-5.10.swift [hunk 18]
```diff
@@ -34,11 +37,18 @@ let package = Package(
             plugins: [
                 .plugin(name: SwiftLint.plugin),
             ]),
+        .target(
+            name: "SnapshotTestingExtensions",
```
Comment (tinder-cfuller) [inline Package@swift-5.10.swift:41]:
```suggestion
            name: "SnapshotTestingExtensions",
            dependencies: [
                .product(name: "SnapshotTesting", package: "swift-snapshot-testing"),
            ],
```

File: Package@swift-5.9.swift [hunk 19]
```diff
@@ -34,11 +37,18 @@ let package = Package(
             plugins: [
                 .plugin(name: SwiftLint.plugin),
             ]),
+        .target(
+            name: "SnapshotTestingExtensions",
```
Comment (tinder-cfuller) [inline Package@swift-5.9.swift:41]:
```suggestion
            name: "SnapshotTestingExtensions",
            dependencies: [
                .product(name: "SnapshotTesting", package: "swift-snapshot-testing"),
            ],
```

**Review by tinder-igornovik — APPROVED** [review]

File: Tests/LayoutTests/Support/SnapshotTesting.swift (hunk 16 above)
Comment (tinder-garricnahapetian) [reply Tests/LayoutTests/Support/SnapshotTesting.swift:13]: It is required:
```
Type 'ViewImageConfig' has no member 'iPhone14'
```

File: Package.swift [hunk 20]
```diff
@@ -23,11 +26,18 @@ let package = Package(
...
+            swiftSettings: .swiftSettings,
+            plugins: [
+                .plugin(name: SwiftLint.plugin),
+            ]),
```
[the remark this answers is not in the document]
Comment (tinder-cfuller) [reply Package.swift]: This change still needs to be made.

File: Sources/SnapshotTestingExtensions/UITraitCollection.swift [hunk 21]
```diff
@@ -0,0 +1,98 @@
...
+//
+
+import SnapshotTesting
+import UIKit
+
+extension UITraitCollection {
+
+    internal static func iPhone14(_ orientation: ViewImageConfig.Orientation) -> UITraitCollection {
+        let base: [UITraitCollection] = [
+            UITraitCollection(forceTouchCapability: .available),
```
Comment (tinder-cfuller) [inline Sources/SnapshotTestingExtensions/UITraitCollection.swift:17]:
```suggestion
            UITraitCollection(forceTouchCapability: .unavailable),
```

File: Sources/SnapshotTestingExtensions/UITraitCollection.swift [hunk 22]
```diff
@@ -0,0 +1,98 @@
...
+            UITraitCollection(traitsFrom: base + [
+                UITraitCollection(horizontalSizeClass: .compact),
+                UITraitCollection(verticalSizeClass: .regular)
+            ])
+        }
+    }
+
+    internal static func iPhone14Plus(_ orientation: ViewImageConfig.Orientation) -> UITraitCollection {
+        let base: [UITraitCollection] = [
+            UITraitCollection(forceTouchCapability: .available),
```
Comment (tinder-cfuller) [inline Sources/SnapshotTestingExtensions/UITraitCollection.swift:38]:
```suggestion
            UITraitCollection(forceTouchCapability: .unavailable),
```

File: Sources/SnapshotTestingExtensions/UITraitCollection.swift [hunk 23]
```diff
@@ -0,0 +1,98 @@
...
+            UITraitCollection(traitsFrom: base + [
+                UITraitCollection(horizontalSizeClass: .compact),
+                UITraitCollection(verticalSizeClass: .regular)
+            ])
+        }
+    }
+
+    internal static func iPhone14Pro(_ orientation: ViewImageConfig.Orientation) -> UITraitCollection {
+        let base: [UITraitCollection] = [
+            UITraitCollection(forceTouchCapability: .available),
```
Comment (tinder-cfuller) [inline Sources/SnapshotTestingExtensions/UITraitCollection.swift:59]:
```suggestion
            UITraitCollection(forceTouchCapability: .unavailable),
```

File: Sources/SnapshotTestingExtensions/UITraitCollection.swift [hunk 24]
```diff
@@ -0,0 +1,98 @@
...
+            UITraitCollection(traitsFrom: base + [
+                UITraitCollection(horizontalSizeClass: .compact),
+                UITraitCollection(verticalSizeClass: .regular)
+            ])
+        }
+    }
+
+    internal static func iPhone14ProMax(_ orientation: ViewImageConfig.Orientation) -> UITraitCollection {
+        let base: [UITraitCollection] = [
+            UITraitCollection(forceTouchCapability: .available),
```
Comment (tinder-cfuller) [inline Sources/SnapshotTestingExtensions/UITraitCollection.swift:80]:
```suggestion
            UITraitCollection(forceTouchCapability: .unavailable),
```

**Review by tinder-cfuller — DISMISSED** [review]

File: Package.swift [hunk 25]
```diff
@@ -23,10 +26,16 @@ let package = Package(
...
+        .target(
+            name: "SnapshotTestingExtensions",
+            dependencies: [
+                .product(name: "SnapshotTesting", package: "swift-snapshot-testing"),
+            ]),
         .testTarget(
             name: "LayoutTests",
             dependencies: [
                 "Layout",
+                "SnapshotTestingExtensions",
```
Comment (tinder-cfuller) [inline Package.swift:38]:
```suggestion
                "SnapshotTestingExtensions",
```

File: Package@swift-5.10.swift [hunk 26]
```diff
@@ -34,10 +37,20 @@ let package = Package(
...
+            dependencies: [
+                .product(name: "SnapshotTesting", package: "swift-snapshot-testing"),
+            ],
```
Comment (tinder-cfuller) [inline Package@swift-5.10.swift:44]:
```suggestion
            ],
```

File: Package@swift-5.10.swift [hunk 27]
```diff
@@ -34,10 +37,20 @@ let package = Package(
...
+            ],            
+            swiftSettings: .swiftSettings,
+            plugins: [
+                .plugin(name: SwiftLint.plugin),
+            ]),
         .testTarget(
             name: "LayoutTests",
             dependencies: [
                 "Layout",
+                "SnapshotTestingExtensions",
```
Comment (tinder-cfuller) [inline Package@swift-5.10.swift:53]:
```suggestion
                "SnapshotTestingExtensions",
```

**Review by tinder-cfuller — COMMENTED** [review]
Some extraneous whitespace snuck in!

File: Package@swift-5.9.swift [hunk 28]
```diff
@@ -34,10 +37,20 @@ let package = Package(
...
+            dependencies: [
+                .product(name: "SnapshotTesting", package: "swift-snapshot-testing"),
+            ],
```
Comment (tinder-cfuller) [inline Package@swift-5.9.swift:44]:
```suggestion
            ],
```

File: Package@swift-5.9.swift [hunk 29]
```diff
@@ -34,10 +37,20 @@ let package = Package(
...
+            ],            
+            swiftSettings: .swiftSettings,
+            plugins: [
+                .plugin(name: SwiftLint.plugin),
+            ]),
         .testTarget(
             name: "LayoutTests",
             dependencies: [
                 "Layout",
+                "SnapshotTestingExtensions",
```
Comment (tinder-cfuller) [inline Package@swift-5.9.swift:53]:
```suggestion
                "SnapshotTestingExtensions",
```

**Review by tinder-cfuller — COMMENTED** [review]
And these as well!

**Review by tinder-cfuller — DISMISSED** [review]

**Comment by tinder-cfuller** [comment]
May you please also change …
```
-scheme "$(target)"
```
… to …
```
-scheme "$(target)-Package"
```
… in the `Makefile` …

https://github.com/Tinder/Layout/blob/main/Makefile#L32
and
https://github.com/Tinder/Layout/blob/main/Makefile#L75

Thank you!

**Comment by tinder-cfuller** [comment]
@tinder-garricnahapetian May you please also add `-Package` to only the `analyze` recipe in the Makefile (leaving the `doc` recipe as is).

https://github.com/Tinder/Layout/blob/main/Makefile#L32

Please change from `-scheme "$(target)"` to `-scheme "$(target)-Package"`

Thank you!

**Review by tinder-cfuller — APPROVED** [review]
👍

**Pull request merged**